### PR TITLE
Check email_opt_in query string in basket url

### PR DESCRIPTION
**Pull request by bbaker6225**
Grab the email_opt_in query string from the basket url if it exists, defaulting to false

### CODE REVIEW

**Comment by jmyatt** [comment]
LGTM, just holding off on the approve/merge to see if there are other values we'll need to stash in this way for the email opt-in functionality

**Comment by bbaker6225** [comment]
Closed as we are looking at other implementations so that ecommerce doesn't have to know about email opt in preferences

**Comment by bbaker6225** [comment]
Open again

**Comment by bbaker6225** [comment]
[Github is dying](https://status.github.com/messages)

Tests should pass once it's working again

File: ecommerce/extensions/basket/tests/test_views.py [hunk 1]
```diff
@@ -265,6 +266,41 @@ def test_with_both_unavailable_and_available_products(self):
...
+        Verify the email_opt_in query string is saved into a BasketAttribute.
+        """
+        url = '{path}?sku={sku}&email_opt_in={opt_in}'.format(
+            path=self.path,
+            sku=self.stock_record.partner_sku,
+            opt_in=opt_in,
+        )
+        self.client.get(url)
+        basket_attribute = BasketAttribute.objects.get(
+            attribute_type=BasketAttributeType.objects.get(name=EMAIL_OPT_IN_ATTRIBUTE),
```
Comment (jmyatt) [inline ecommerce/extensions/basket/tests/test_views.py:287]: I think you want to pass the basket into these BasketAttribute get's, no?

**Review by jmyatt — APPROVED** [review]
LGTM - please squash

**Pull request merged**